DigiMeSDK/Core/Classes/DMENativeConsentManager.m

```diff
@@ -65,6 +66,10 @@ - (void)handleAction:(DMEOpenAction *)action withParameters:(NSDictionary<NSStri
     {
         error = [NSError authError:AuthErrorGeneral];
     }
+    else if (reference != nil)
```

Comment:
Looking at the code, this condition will never actually be executed. When `reference` is present in the callbackUrl, the `result` will be set to `kDMEResultValueError`, right? Which means it will be picked up by the earlier `else if`.

Reply:
If you combine this comment with the one about `AuthErrorGeneral`, you could probably re-write this to somthing like:

```swift
else if ([result isEqualToString:kDMEResultValueError])
    {
        error = [NSError authError:AuthErrorGeneral reference:reference]; //where `reference` can be `nullable`?
    }
```